### PR TITLE
feat(oga): show OGA action response in UI

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary

Add support for displaying the OGA officer’s action response in the UI after a task is completed, so previously submitted reviewer actions can be viewed later instead of disappearing after submission.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Changes Made

- Updated the backend application payload to expose both the original submitted data and the OGA action response separately.
- Changed the application detail flow to load a read-only data form for submitted information and a separate OGA action form for reviewer actions.
- Updated the frontend workflow detail screen to render the OGA response data and submit the correct action payload.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

- No issues created

## Additional Notes

This change keeps the applicant’s submitted data visible while also preserving the OGA officer’s completed action response for later review.

### CODE REVIEW

File: portals/apps/oga-app/src/api.ts [hunk 1]
```diff
@@ -113,11 +113,16 @@ export interface OGAApplication {
...
   };
-  form: {
+  dataForm?: {
+    schema: JsonSchema;
+    uiSchema: UISchemaElement;
+  };
+  ogaForm: {
     schema: JsonSchema;
     uiSchema: UISchemaElement;
   };
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply portals/apps/oga-app/src/api.ts]: No there will be always a default form will be there at `oga/data/forms/default.json`

File: portals/apps/oga-app/src/screens/WorkflowDetailScreen.tsx [hunk 2]
```diff
@@ -78,7 +78,8 @@ export function WorkflowDetailScreen() {
       try {
         const data = await fetchApplicationDetail(apiClient, taskId)
         setApplication(data)
-        setFormConfig({ schema: data.form.schema, uiSchema: data.form.uiSchema })
+        setOgaFormConfig({ schema: data.ogaForm.schema, uiSchema: data.ogaForm.uiSchema })
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply portals/apps/oga-app/src/screens/WorkflowDetailScreen.tsx]: Addressed both comments! Made `ogaForm` optional in the interface and added a null guard before using it. Fail-safe is now engaged.

**Review by ginaxu1 — APPROVED** [review]

File: portals/apps/oga-app/src/api.ts [hunk 3]
```diff
@@ -113,11 +113,16 @@ export interface OGAApplication {
...
   meta?: {
     type: string;
     verificationId: string;
   };
-  form: {
+  dataForm?: {
+    schema: JsonSchema;
+    uiSchema: UISchemaElement;
+  };
+  ogaForm?: {
```
Comment (mushrafmim) [inline portals/apps/oga-app/src/api.ts:125]: Isn't `data` the part of the jsonforms, shouldn't we move the `data` field and the `ogaActionData` field inside `dataForm` and `ogaForm`? I think this is the setup in `NSW` side, but missing here even in the previous implementation.

**Review by mushrafmim — DISMISSED** [review]

File: portals/apps/oga-app/src/api.ts (hunk 3 above)
Comment (sthanikan2000) [reply portals/apps/oga-app/src/api.ts:125]: Data can be come as markdown as well.  Once we implement the template handler. We should make this definitions something like
```json
data: {
    "template":
    "type":
    "content":
}
```
This should be done as separate PR, once we implement the Template handler

**Review by Aravinda-HWK — APPROVED** [review]
I checked this PR locally. LGTM
